src/librustc/diagnostics.rs

```diff
@@ -1359,9 +1359,10 @@ how long the data stored within them is guaranteed to be live. This lifetime
 must be as long as the data needs to be alive, and missing the constraint that
 denotes this will cause this error.
 
+This won't compile because T is not constrained, meaning the data stored in it
+is not guaranteed to last as long as the reference.
+
 ```compile_fail,E0309
-// This won't compile because T is not constrained, meaning the data
```

Comment:
The explanation is actually better as a comment (at least for me) because outside of the block, it looks strange... :-/